README.md

```diff
@@ -223,12 +223,18 @@ Instead, it relies on LVM2 to lock around operations that are not reentrant.
 
 #### Logical volume naming
 
-The volume group name is specified at startup through the
-`-volume-group` argument.
+The volume group name is specified at startup through the `-volume-group` argument.
 
-Logical volumes are named according to the following pattern
-`<volume-group-name>_<logical-volume-name>`.
+Logical volumes names are chosen from randomly-generated, base36-encoded numbers.
+The CO-specified volume name is captured in a LV tag: `<prefix>_<fn(CO-specified-name)>`, where `<prefix>` is either of:
```

Comment:
It's a bit hard for me to understand this description. How about:

The CO-specified volume name is captured in an LV tag conforming to one of the following format:
* `VN.<CO-specified-name>`, if the CO-specified name contains *only* characters safe for LVM tags (`a-z A-Z 0-9 + _ . -`).
* `VN+<base64-rawurl-encode(CO-specified-name)>`, otherwise.

The underscore between the prefix and the CO-specified-name should not be there, according to the  example below.